test/parallel/test-buffer-slice.js

```diff
@@ -62,3 +62,15 @@ assert.strictEqual(Buffer.alloc(0).slice(0, 1).length, 0);
 
 // slice(0,0).length === 0
 assert.strictEqual(0, Buffer.from('hello').slice(0, 0).length);
+
+{
+  // Test included in https://github.com/nodejs/node/issues/9096
+  // If the offset or ending value provided is a floating point number, it
+  // should be coerced to integer consistently. Otherwise the actual length of
+  // the slice calculation will be messed up.
```

Comment:
You mean to say the slice implementation would be better done in C++?

Honestly, I would prefer to throw an error if the input is not an integer, which would avoid all this.